app/controllers/wpcc/your_contributions_controller.rb

```diff
@@ -3,17 +3,15 @@
 module Wpcc
   class YourContributionsController < EngineController
     protect_from_forgery
+    after_action :store_eligible_salary, only: :new
```

Comment:
I agree that it being only on one other method is not the point, however the description 'something to be handle on other steps' makes that implication.

It is part of the new action, there's no other place it can occur. By including it in the new action, it is self-evident to anyone coming along new to the code that something needs to happen regarding sessions. Since we are not using a d/b, we need to make clear that we are relying on the session to persist the data.